### PR TITLE
Add missing read unlock calls in transit backend code

### PR DESCRIPTION
**Pull request by divyapola5**
This PR is a followup for #12418 and in summary includes the following changes:
1 - Add missing read unlock calls for cacheSizeChanged variable
2 - Invoke initCache in pathCacheConfigRead routine if needed

### CODE REVIEW

File: changelog/12652.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+```release-note:bug
+Add missing read unlock calls in transit backend code.
```
Comment (victorr) [inline changelog/12652.txt:2]: I don't have experience writing changelog entries but it seems to me that as the audience are Vault users/operators, you don't need to mention detailed information such as methods called etc. Perhaps something more generic along the lines of: fixed a problem with the transit cache, and made it unnecessary to reload the backend when the configuration changes.

File: builtin/logical/transit/backend.go [hunk 2]
```diff
@@ -122,11 +122,13 @@ func (b *backend) GetPolicy(ctx context.Context, polReq keysutil.PolicyRequest,
 		currentCacheSize := b.lm.GetCacheSize()
 		storedCacheSize, err := GetCacheSizeFromStorage(ctx, polReq.Storage)
 		if err != nil {
+			b.configMutex.RUnlock()
```
Comment (victorr) [inline builtin/logical/transit/backend.go:125]: These code changes are 🆗, but please consider making the equivalent changes as you did in the Transform plugin (extracting a method, changing the if statement into a guard).

**Review by victorr — APPROVED** [review]
The changes look good to me, but please verify the changelog entry with someone (maybe Scott?)

File: changelog/12652.txt (hunk 1 above)
Comment (sgmiller) [reply changelog/12652.txt:2]: We probably already have the changelog entry for the overall cache issue, so I think we can just remove this.

**Review by sgmiller — CHANGES_REQUESTED** [review]

File: builtin/logical/transit/backend.go (hunk 2 above)
Comment (sgmiller) [reply builtin/logical/transit/backend.go:125]: I'd like to get this merged ASAP so as not to block the point releases.  Divya: can you follow up sometime after with a PR that aligns this to the transform changes?

File: changelog/12652.txt (hunk 1 above)
Comment (divyapola5) [reply changelog/12652.txt]: Thank you. Removed the changelog entry for this

File: builtin/logical/transit/backend.go (hunk 2 above)
Comment (divyapola5) [reply builtin/logical/transit/backend.go:125]: Sure. I will open a new PR for new helper method.

**Review by sgmiller — APPROVED** [review]

**Pull request merged**